### ISSUE TITLE
[BUG]: Metal Gear Solid 3: Subsistence wrong internal framerate

### ISSUE THREAD

**Issue by HelicopterP** [PCSX2/pcsx2#6351]
### Describe the Bug

PCSX2 renders this game at 60 internal FPS all the time, while real PS2 runs it at 30fps with lots of slowdowns to 15-20fps during cutscenes.
You can see a real PS2 performance in this video by Digital Foundry here https://www.youtube.com/watch?v=poEqykRrxDk
alternatively you can observe it by eye here https://www.youtube.com/watch?v=eytZvbL4iwQ&t=1524s
What this means in practice is instead of emulating slowdown of a real PS2 while outputing 60 Hz and playing audio at the correct speed PCSX2 tries to power past this issue rendering everything at 60 fps resulting in VERY HIGH system requirements (especially in software rendering mode)
![MGS3](https://user-images.githubusercontent.com/4709459/172067070-d2d03829-d91c-450b-a48f-964445521724.jpg)
.

### Reproduction Steps

Start the game and observe framerate. 

### Expected Behavior

Emulation of a framerate of a real PS2 - 30fps with drops to 15-20fps while maintaining 60 vertical interrupts. 

### PCSX2 Revision

v1.7.2893

### Operating System

Windows 10 (64bit)

### CPU

I7-8700

### GPU

GTX1070

### Emulation Settings

All speedhacks disabled except for instant VU1.

**Comment by refractionpcsx2** [comment]
Lower the EE Cycle Rate slider then :)

This isn't something we can easily fix, without making the emulation even HARDER on your computer, because it means we would have to calculate and time bus busy times, constantly check the state of other components from inside the EE JIT and exit the EE JIT (Super slow to do this) emulate the cache to bypass bus contention (EXTREEEEEEMELY slow) and probably a bunch of other shit, just to make it run world (more PS2-like) internally.

So no, sorry it's not going to happen.  Either gimp it with lowering the cycle rate or suffer with your better framerates, i guess..


I can now thankfully mark "Somebody complaining about a better experience" off my bucket list.

**Comment by refractionpcsx2** [comment]
Okay it's just come to my attention that this can be altered by preferring the Blitting method of detection, as expected the game does things to screw us over, so this might be something changed soon.

**Comment by HelicopterP** [comment]
Well, in this case faster internal framerate means higher system requirements anyway, because getting fps 3 times higher than a real PS2 requires faster CPU and GPU. Not as high as emulating cache and stuff obviously, but who knows whether it's really needed...

What is also interesting is that some games (like Jak 2 for example) can slow down internally while PCSX2 maintains 60 VIs.

I am not relying on a internal framecounter, the screenshot is for illustration purposes only. The difference between 60 fps and 20 fps is clearly visible without any tools. Given hardware fast enough MGS3 will run at 60 internal FPS all the time in PCSX2. I am actually quite surprised that it works without any other glitches at this speed.

There is also no Cycle rate slider in QT :p OK, ok, joking :D

**Comment by refractionpcsx2** [comment]
Well I tested MGS3, and with Blitting as a method of internal framerate detection, it said 25 (on a PAL 50fps version), which is closer to what you are seeing on that PS2 footage.  so the difference is more likely to be 30fps vs 20fps, rather than 60fps vs 20fps

**Comment by HelicopterP** [comment]
Ok, I had a bit more time to play with this and here are my findings.
1. MGS3 most definatelly runs at internal 60 fps in PCSX2 during cutscenes. It is clearly visible by the eye, but I double checked myself just in case. Here is a short clip captured by PCSX2, it is a 59.94 fps recording obviously.  You can play it in VLC then pause it and press E key several times (by each press of E key you are advancing playback by 1 frame). You will see that each new frame is different thus proving that game is indeed running at 60 fps. (It would've showed one new frame followed by a duplicate frame if it was running at 30 fps, which is not the case here). 
[MGSTEST.zip](https://github.com/PCSX2/pcsx2/files/8864194/MGSTEST.zip)
2. Gameplay is running at interanal 30 fps. I did not test it during my original report, because It was too time-consuming to get to gameplay and I did not expect different results. This means that game FPS counter is wrong in PCSX2 in this case (but right during cutscenes), but this is not the point.

After discovering this I thought that this game is double buffered and because it can not maintain 60 fps on real hardware it runs at 30fps and when it can't hit 30 it drops to 20. To test this theory I increased EE cyclerate to 300% aaand… nothing changed - gameplay was still limited to 30fps. Apparently the the problem is somewhere else… Maybe MGS3 uses different methods for fps limiting during cutscenes and gameplay…

TL;dr original report was correct and it is a valid problem, PCSX2 runs this game several times faster than it should. Whether it could be fixed without totally ruining performance in general or not is not up to me to decide and debate obviously.
